metadata_backend/helpers/metax_mapper.py

```diff
@@ -0,0 +1,443 @@
+                            "type": "object",
+                            "$ref": "#/definitions/Concept",
+                        },
+                    },
+                },
+            },
+            # TODO: will be implemented later
+            "field_of_science": {
+                "type": "array",
+                "items": {"type": "object", "$ref": "#/definitions/Concept"},
+            },
+            # TODO: Need clarification on necessarity of this field
+            "remote_resources": {
+                "type": "array",
+                "items": {"type": "object", "$ref": "#/definitions/WebResource"},
+            },
+            # restricted
+            "access_rights": {
+                "type": "object",
+                "$ref": "#/definitions/RightsStatement",
+            },
+            # contributors - Data Curator
+            "curator": {
+                "type": "array",
+                "items": {"type": "object", "$ref": "#/definitions/ResearchAgent"},
+            },
+            # TODO: will be implemented later
+            "total_remote_resources_byte_size": {
+                "type": "integer",
+            },
+        }
+    }
+
+    def __init__(self, metax_data: Dict, data: Dict) -> None:
+        """Set variables.
+
+        :param metax_data: Metax research_dataset metadata
+        """
+        self.research_dataset = metax_data
+        self.datacite_data = data
+        self.identifier_types = metax_reference_data["identifier_types"]
+        self.person: Dict[str, Any] = {
+            "name": "",
+            "@type": "Person",
+            "member_of": {"name": {"en": ""}, "@type": "Organization"},
+            "identifier": "",
+        }
+
+    def map_metadata(self) -> Dict[str, Any]:
+        """Public class for actual mapping of metadata fields.
+
+        :returns: Research dataset
+        """
+        LOG.info("Mapping datasite data to Metax metadata")
+        LOG.debug("Data incomming for mapping: ", self.datacite_data)
+        for key, value in self.datacite_data.items():
+            if key == "creators":
+                self._map_creators(value)
+            if key == "keywords":
+                self.research_dataset["keyword"] = value.split(",")
+            if key == "contributors":
+                self._map_contributors(value)
+            if key == "dates":
+                self._map_dates(value)
+            if key == "geoLocations":
+                self._map_spatial(value)
+            if key == "alternateIdentifiers":
+                self._map_other_identifier(value)
+        return self.research_dataset
+
+    def _map_creators(self, creators: List) -> None:
+        """Map creators.
+
+        :param creators: Creators data from datacite
+        """
+        LOG.info("Mapping creator")
+        LOG.debug(creators)
+        self.research_dataset["creator"] = []
+        for creator in creators:
+            metax_creator = deepcopy(self.person)
```

Comment:
not sure a `deepcopy` is needed here, i suspect a shallow copy would suffice. 
Can I get more information on the intent here?

Reply:
if shallow copied the next instance of `self.person` won't have e.g. identifier field because it could be deleted. Of course, this can be implemented differently...

Reply:
ok, got it, then we can leave as is.